demoextendtemplates/src/Controller/DecoratedOrderController.php

```diff
@@ -49,10 +49,220 @@ public function __construct(
+    {
+        return $this->orderController->addPaymentAction($orderId, $request);
+    }
+
+    public function previewAction(int $orderId)
+    {
+        return $this->orderController->previewAction($orderId);
+    }
+
+    public function duplicateOrderCartAction(int $orderId)
+    {
+        return $this->orderController->duplicateOrderCartAction($orderId);
+    }
+
+    public function sendMessageAction(Request $request, int $orderId)
+    {
+        return $this->orderController->sendMessageAction($request, $orderId);
+    }
+
+    public function changeCustomerAddressAction(Request $request)
+    {
+        return $this->orderController->changeCustomerAddressAction($request);
+    }
+
+    public function changeCurrencyAction(int $orderId, Request $request)
+    {
+        return $this->orderController->changeCurrencyAction($orderId, $request);
+    }
+
+    public function resendEmailAction(int $orderId, int $orderStatusId, int $orderHistoryId)
+    {
+        return $this->orderController->resendEmailAction($orderId, $orderStatusId, $orderHistoryId);
+    }
+
+    public function deleteProductAction(int $orderId, int $orderDetailId)
+    {
+        return $this->orderController->deleteProductAction($orderId, $orderDetailId);
+    }
+
+    public function getDiscountsAction(int $orderId)
+    {
+        return $this->orderController->getDiscountsAction($orderId);
+    }
+
+    public function getPricesAction(int $orderId)
+    {
+        return $this->orderController->getPricesAction($orderId);
+    }
+
+    public function getPaymentsAction(int $orderId)
+    {
+        return $this->orderController->getPaymentsAction($orderId);
+    }
+
+    public function getProductsListAction(int $orderId)
+    {
+        return $this->orderController->getProductsListAction($orderId);
+    }
+
+    public function generateInvoiceAction(int $orderId)
+    {
+        return $this->orderController->generateInvoiceAction($orderId);
+    }
+
+    public function sendProcessOrderEmailAction(Request $request)
+    {
+        return $this->orderController->sendProcessOrderEmailAction($request);
+    }
+
+    public function cancellationAction(int $orderId, Request $request)
+    {
+        return $this->orderController->cancellationAction($orderId, $request);
+    }
+
+    public function configureProductPaginationAction(Request $request)
+    {
+        return $this->orderController->configureProductPaginationAction($request);
+    }
+
+    public function displayCustomizationImageAction(int $orderId, string $value)
+    {
+        return $this->orderController->displayCustomizationImageAction($orderId, $value);
+    }
+
+    public function setInternalNoteAction(int $orderId, Request $request)
+    {
+        return $this->orderController->setInternalNoteAction($orderId, $request);
+    }
+
+    public function searchProductsAction(Request $request)
+    {
+        return $this->orderController->searchProductsAction($request);
+    }
```

Comment:
so we must replace all of the actions even if we don't change anything in most of them? So tidious :(
Shouldn't we extend the original controller instead?  All the public actions should work by default at least (Im not sure if its a good practice, but if it works at least for such scenarios it seems to fit better). @FabienPapet wdyt?

Reply:
as we spoke, it will get a dedicated example on how to work with overrides of services, how to remap routing, etc.